### PR TITLE
azure_rm_cosmosdbaccount - fix for virtual network rules

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Fix for https://github.com/ansible/ansible/issues/52138
Wrong version of azure-mgmt.-cosmosdb package was referred.
In addition added appropriate test and minor fix in suboption definition.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_cosmosdbaccount

### CODE REVIEW

File: test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml [hunk 1]
```diff
@@ -3,6 +3,24 @@
     dbname: "cosmos{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
   run_once: yes
 
+- name: Create virtual network
+  azure_rm_virtualnetwork:
+    resource_group: "{{ resource_group }}"
+    name: testvn
```
Comment (yungezz) [inline test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]: random name?

File: test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml [hunk 2]
```diff
@@ -3,6 +3,24 @@
...
+    address_prefixes_cidr:
+      - 10.1.0.0/16
+      - 172.100.0.0/16
+    dns_servers:
+      - 127.0.0.1
+      - 127.0.0.3
+
+- name: Add subnet
+  azure_rm_subnet:
+    name: testsubnet
```
Comment (yungezz) [inline test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]: random name?

File: test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml (hunk 1 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]: yes, I will add it here

File: test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml (hunk 2 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml]: i think not necessary for subresource

**Review by yungezz — APPROVED** [review]

**Pull request merged**